nodeJS/express/using_postgresql.md

```diff
@@ -0,0 +1,309 @@
+<!-- markdownlint-disable TOP004 -->
+
+### Introduction
+
+Now that we have PostgreSQl up and running on our machine, it's time to work with it.
+
+<div class="lesson-note lesson-note--critical" markdown="1">
+
+#### Prerequisite SQL knowledge
+
+Make sure you've completed the [SQL course](https://www.theodinproject.com/paths/full-stack-javascript/courses/databases).
+
+This lesson and all subsequent lessons will assume you understand SQL syntax and concepts.
+
+</div>
+
+For brevity's sake, we'll refer to database as db from now on.
+
+### Lesson overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+- Setting up a new db in PostgreSQL shell
+- Setting up and querying with node-postgres
+- Using a script to populate the db
+
+### Setting up a Express app
+
+Let's start by creating a Express application. It will just have one feature - add usernames provided by the user to the db. Here are the expected routes and their functionalities:
+
+- `GET /` - to log available usernames in the DB to the terminal. For now, just put in a simple `console.log('usernames will be logged here - wip')`.
+- `GET /new` - to display a HTML form to the user with one username input text field. It will submit to the following route,
+- `POST /new` - will save the incoming username data to the DB. For now, just log `console.log("username to be saved: ", req.body.username)`.
+
+Move on to the next section once you have the above functionality working. Scope relevant code to the `routes` and `controllers` folders. Since there's only one view (`GET /new`) to take care of, it's up to you whether you want to use `ejs` or make do with plain ol' HTML.
```

Comment:
Here and line 106, should we change folder/folders to directory/directories?

Reply:
is there a particular reason to switch?

Reply:
My understanding was that "directory" is more technically correct as "folder" technically refers to the GUI representation of a directory. But having said that, it looks like we use "folder" a lot throughout the curriculum anyway, so no biggie at all.